### PR TITLE
Change addSplit in SourceOperator to take ScheduledSplit

### PR DESCRIPTION
**Pull request by miaoever**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/ExchangeOperator.java [hunk 1]
```diff
@@ -115,8 +116,10 @@ public PlanNodeId getSourceId()
     }
 
     @Override
-    public Supplier<Optional<UpdatablePageSource>> addSplit(Split split)
+    public Supplier<Optional<UpdatablePageSource>> addSplit(ScheduledSplit scheduledSplit)
     {
+        requireNonNull(scheduledSplit, "scheduledSplit is null");
+        Split split = scheduledSplit.getSplit();
         requireNonNull(split, "split is null");
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/operator/ExchangeOperator.java:123]: nit: these 3 lines can be merged into one:
```java
Split split = requireNonNull(scheduledSplit, "scheduledSplit is null").getSplit();

File: presto-main/src/main/java/com/facebook/presto/operator/MergeOperator.java [hunk 2]
```diff
@@ -152,8 +153,10 @@ public PlanNodeId getSourceId()
     }
 
     @Override
-    public Supplier<Optional<UpdatablePageSource>> addSplit(Split split)
+    public Supplier<Optional<UpdatablePageSource>> addSplit(ScheduledSplit scheduledSplit)
     {
+        requireNonNull(scheduledSplit, "scheduledSplit is null");
+        Split split = scheduledSplit.getSplit();
         requireNonNull(split, "split is null");
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/operator/MergeOperator.java:160]: same

File: presto-main/src/main/java/com/facebook/presto/operator/ScanFilterAndProjectOperator.java [hunk 3]
```diff
@@ -135,8 +136,10 @@ public PlanNodeId getSourceId()
     }
 
     @Override
-    public Supplier<Optional<UpdatablePageSource>> addSplit(Split split)
+    public Supplier<Optional<UpdatablePageSource>> addSplit(ScheduledSplit scheduledSplit)
     {
+        requireNonNull(scheduledSplit, "scheduledSplit is null");
+        Split split = scheduledSplit.getSplit();
         requireNonNull(split, "split is null");
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/operator/ScanFilterAndProjectOperator.java:143]: same and everywhere else in the PR

**Review by highker — COMMENTED** [review]
nits only

**Comment by miaoever** [comment]
@highker Addressed the comments. (There are some test failures from presto-native-execution not related to this change.)

**Review by highker — APPROVED** [review]

**Pull request merged**